js/bootstrap-datepicker.js

```diff
@@ -1112,7 +1112,7 @@
 
 		keydown: function(e){
 			if (this.picker.is(':not(:visible)')){
-				if (e.keyCode === 27) // allow escape to hide and re-show picker
+				if (e.keyCode === 40) // allow down to re-show picker
```

Comment:
Can you please, like @eternicode suggested, add the escape key back. That way we don't break BC ;)